pkgs/tools/typesetting/tex/nix/default.nix

```diff
@@ -63,17 +64,20 @@ rec {
             { rootFile = baseNameOf (toString rootFile); src = key; }
             "${pkgs.perl}/bin/perl ${./find-includes.pl}");
 
+          searchPath' = [(dirOf key)] ++ searchPath;
+
           # Look for the dependencies of `key', trying various
           # extensions determined by the type of each dependency.
-          # TODO: support a search path.
           foundDeps = dep: xs:
             let
               exts =
                 if dep.type == "img" then [".pdf" ".png" ".ps" ".jpg"]
                 else if dep.type == "tex" then [".tex" ""]
                 else [""];
               fn = pkgs.lib.findFirst (fn: builtins.pathExists fn) null
-                (map (ext: dirOf key + ("/" + dep.name + ext)) exts);
+                (builtins.concatLists
+                  (map (path: map (ext: toString "${path}/${dep.name}${ext}") exts) searchPath')
```

Comment:
Why do you have `toString` here?

Also `"${path}/${file}"` is not the same as `toString (path + "/${file}")`, is that change intentional?

Reply:
Removing the `toString` from line 79 seems to not break things, although I am not 100% confident that will always be the case.

removing the `toString` from line 51 definitely breaks things. (At least with the input I tried it with)

Reply:
> Is there a place I can read about what the difference is?

Sorry, didn't mean to be cryptic. The "${path}/${file}" will first put the whole "path" directory into the nix store and the resulting string will point to a file in that store directory. The `path + "/${file}"` doesn't try to reduce "path" itself to a string, the result will be still a path (in nixlang terms), so that allows for just a file to be placed into the store.